packages/wonder-blocks-button/components/button-core.js

```diff
@@ -0,0 +1,271 @@
+// @flow
+import React from "react";
+import {StyleSheet, css} from "aphrodite";
+
+import {LabelLarge} from "wonder-blocks-typography";
+import {processStyleList} from "../../wonder-blocks-core/util/util.js";
+import Color, {mix, fade} from "wonder-blocks-color";
+import type {ValidTints} from "wonder-blocks-color";
+import type {SharedProps} from "../types.js";
+
+type Props = SharedProps & {
+    hovered?: boolean,
+    focused?: boolean,
+    pressed?: boolean,
+};
+
+export default class ButtonCore extends React.Component<Props> {
+    render() {
+        const {
+            color,
+            kind,
+            light,
+            size,
+            disabled,
+            hovered,
+            focused,
+            pressed,
+        } = this.props;
+
+        const buttonStyles = _generateStyles(
+            color || Color.blue,
+            kind || "primary",
+            !!light,
+        );
+
+        if (disabled) {
+            return (
+                <button
+                    className={css(
+                        sharedStyles.shared,
+                        sharedStyles.disabled,
+                        size === "small" && sharedStyles.small,
+                        buttonStyles.default,
+                        buttonStyles.disabled,
+                    )}
+                    disabled
+                >
+                    <LabelLarge>{this.props.children}</LabelLarge>
+                </button>
+            );
+        }
+        return (
+            <button
```

Comment:
@kevinbarabash It feels like we should be using a View here to be able to support easy styling - however View doesn't support custom tags. Should that be changed?

Reply:
That's an interesting idea.  It seems like that would could in pretty useful.  I wonder if we should have `View` also clear the styling on those tags. 🤔